subsys/dfu/img_util/flash_img.c

```diff
@@ -52,6 +57,63 @@ static bool flash_verify(const struct flash_area *fa, off_t offset,
 	return (len == 0) ? true : false;
 }
 
+#ifdef CONFIG_IMG_ERASE_PROGRESSIVELY
+
+static int flash_sector_from_off(struct flash_area const *fap, off_t off,
+				 struct flash_sector *sector)
+{
+	struct flash_pages_info page;
+	struct device *flash_dev;
+	int rc = -ENODEV;
+
+	flash_dev = flash_area_get_device(fap);
+	if (flash_dev) {
+		rc = flash_get_page_info_by_offs(flash_dev, off, &page);
+		if (rc == 0) {
+			sector->fs_off = page.start_offset;
+			sector->fs_size = page.size;
+		}
+	}
+
+	return rc;
+}
+
+/**
+ * Erase the image slot progressively
+ *
+ * This function erases a flash page to which offset belongs if only this page
+ * wasn't erased before.
+ *
+ * @param[in] ctx contex of the image collection process.
+ * @param[in] off offset from the beginning of the image flash area beginning
+ *
+ * @return  0 on success, negative errno code on fail.
+ */
+static int flash_progressive_erase(struct flash_img_context *ctx, off_t off)
+{
+	struct flash_sector sector;
+	int rc;
+
+	rc = flash_sector_from_off(ctx->flash_area, off, &sector);
+	if (rc) {
+		LOG_ERR("Unable to determine flash sector size");
+	} else {
+		if (ctx->off_last != sector.fs_off) {
+			ctx->off_last = sector.fs_off;
+			LOG_INF("Erasing sector at offset 0x%x", sector.fs_off);
+			rc = flash_area_erase(ctx->flash_area, sector.fs_off,
+					      sector.fs_size);
+			if (rc) {
+				LOG_ERR("Error %d while erasing sector", rc);
+			}
+		}
+	}
+
+	return rc;
+}
+
```

Comment:
excessive empty line

Reply:
prefer to keep, for me it improve legibility